app/controllers/pieces_controller.rb

```diff
@@ -23,16 +23,19 @@ def update
     if @piece.move_to!(new_x, new_y)
       if @piece.game.checkmate?(@piece.game.kings.where.not(color: @piece.color).first) # current player placed other player in checkmate - wins!
         @piece.game.update_attributes(game_winner: @piece.user_id) # set game winner
-        if @piece.game.game_winner == @piece.game.black_user_id
-          winner = "Black"
+        flash[:alert] = "Checkmate! You win!" # will only get set and display on winner's turn
+      else
+        # check for stalemate of other player after move and set game model field
+        opponent_king = @piece.game.kings.where.not(color: @piece.color).first
```

Comment:
Seems to me if this opponent_king variable were set higher in the conditional logic ladder it could be used as an argument multiple times/places. But that is a potential DRY, not a functional necessity.